Algorithm.Framework/Selection/QC500UniverseSelectionModel.cs

```diff
@@ -66,20 +66,26 @@ public override IEnumerable<Symbol> SelectCoarse(QCAlgorithm algorithm, IEnumera
                 return Universe.Unchanged;
             }
 
-            // The stocks must have fundamental data
-            // The stock must have positive previous-day close price
-            // The stock must have positive volume on the previous trading day
             var sortedByDollarVolume =
                 (from x in coarse
                  where x.HasFundamentalData && x.Volume > 0 && x.Price > 0
                  orderby x.DollarVolume descending
                  select x).Take(_numberOfSymbolsCoarse).ToList();
 
             _dollarVolumeBySymbol.Clear();
-            foreach (var i in sortedByDollarVolume)
+            foreach (var x in sortedByDollarVolume)
             {
-                _dollarVolumeBySymbol[i.Symbol] = i.DollarVolume;
+                _dollarVolumeBySymbol[x.Symbol] = x.DollarVolume;
             }
+
+            if (_dollarVolumeBySymbol.Count == 0)
+            {
+                algorithm.Debug($@"QC500UniverseSelectionModel.SelectCoarse: Since no security has met the QC500 criteria,
+the current universe is unchanged. A new selection will be attempted on the next trading day.
+CoarseFundamental Count: {coarse.Count()}");
+                return Universe.Unchanged;
```

Comment:
Nice 👍

Reply:
Logic is fine but please make it silent; not need to debug this.